/*
 * Copyright (C) 2020 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

package gnss_grpc_proxy;

option java_multiple_files = true;
option java_package = "com.android.cuttlefish.gnssproxy.proto";

// The greeting service definition.
service GnssGrpcProxy {
  // Sends NmeaRequest
  rpc SendNmea (SendNmeaRequest) returns (SendNmeaReply) {}

  // Sends GpsRequest
  rpc SendGps (SendGpsRequest) returns (SendGpsReply) {}

  //// Sends GPS vector of data
  rpc SendGpsVector (SendGpsCoordinatesRequest) returns (SendGpsCoordinatesReply) {}
}


// The request message containing nmea
message SendNmeaRequest {
  string nmea = 1;
}

// The response message containing the return nmea reply message
message SendNmeaReply {
  string reply = 1;
}


// The request message containing gps location information
message SendGpsRequest {
  string gps = 1;
}

// The response message containing the return information
message SendGpsReply {
  string reply = 1;
}

message GpsCoordinates {
  float latitude = 1;
  float longitude = 2;
  float elevation = 3;
}

// The request message containing array of gps locations
message SendGpsCoordinatesRequest {
  //Delay in millisecond
  int32 delay =1;
  repeated GpsCoordinates coordinates = 2;
}

// The response message containing the return status or error code if exists
message SendGpsCoordinatesReply {
  enum StatusCode {
    OK = 0;
    CANCELLED = 1;
    UNKNOWN = 2;
    INVALID_ARGUMENT = 3;
    DEADLINE_EXCEEDED = 4;
    NOT_FOUND = 5;
    ALREADY_EXISTS = 6;
    PERMISSION_DENIED = 7;
    RESOURCE_EXHAUSTED = 8;
    FAILED_PRECONDITION = 9;
    ABORTED = 10;
    OUT_OF_RANGE = 11;
    UNIMPLEMENTED = 12;
    INTERNAL = 13;
    UNAVAILABLE = 14;
    DATA_LOSS = 15;
    UNAUTHENTICATED = 16;
  }

  StatusCode status = 1;
}